bindings/pydeck/pydeck/bindings/providers.py

```diff
@@ -1,4 +1,20 @@
-# List of base-map providers
+from pydeck.exceptions import PydeckException
+
+
 class Providers:
+    """Basemap providers available in pydeck"""
+
     MAPBOX = "mapbox"
     GOOGLE_MAPS = "google_maps"
+    CARTO = "carto"
+
+    @classmethod
+    def as_list(cls):
+        return [getattr(cls, a) for a in dir(cls) if not a.startswith("__") and not callable(getattr(cls, a))]
+
+    @classmethod
+    def in_list_or_raise(cls, value):
+        if value not in cls.as_list():
+            raise PydeckException(
+                f"{value} is not a recognized API key provider. Choose one of the following: {cls.as_list()}"
+            )
```

Comment:
I'd suggest subclassing an `Enum` instead, then you don't have to define these class methods...

```py
from enum import Enum
class Providers(str, Enum):
    """Basemap providers available in pydeck"""
    MAPBOX = "mapbox"
    GOOGLE_MAPS = "google_maps"
    CARTO = "carto"
```

Then to get the list of values you can just do 
```py
[x.value for x in Providers]
# ['mapbox', 'google_maps', 'carto']
[x.name for x in Providers]
# ['MAPBOX', 'GOOGLE_MAPS', 'CARTO']
```